pkg/codegen/go/gen_program.go

```diff
@@ -145,21 +146,34 @@ func GenerateProject(directory string, project workspace.Project, program *pcl.P
 	}
 	files["Pulumi.yaml"] = projectBytes
 
+	packages, err := programPackageDefs(program)
+	if err != nil {
+		return err
+	}
+
 	// Build a go.mod based on the packages used by program
 	var gomod bytes.Buffer
 	gomod.WriteString("module " + project.Name.String() + "\n")
+	gomod.WriteString("go 1.17\n")
+	for name, sdk := range localProjects {
+		if name == "go" {
+			gomod.WriteString(fmt.Sprintf("replace github.com/pulumi/pulumi/sdk/v3/go => %s\n", sdk))
+		} else {
+			for _, p := range packages {
+				if p.Name == name {
+					gomod.WriteString(fmt.Sprintf("replace github.com/pulumi/pulumi-%s/sdk/go => %s/go\n", name, sdk))
+				}
+			}
+		}
+	}
+
 	gomod.WriteString(`
-go 1.17
 
 require (
 	github.com/pulumi/pulumi/sdk/v3 v3.30.0
```

Comment:
Oh god's this is hard coded to 3.30! Pretty sure this should be floating latest like for dotnet.